contributors/design-proposals/api-machinery/customresource-conversion-webhook.md

```diff
@@ -125,21 +125,41 @@ type CustomResourceConversionWebhook {
 }
 ```
 
-### Defaulting
+### Top level fields to Per-Version fields
```

Comment:
it might be clearer to start with the v1beta1 and v1 structures and rules, and move this discussion into an "Alternatives considered" section

v1beta1 structure:
* top-level and per-version schema/subresource/printer fields

v1beta1 rules:
* top-level and per-version fields are mutually exclusive (attempting to set both will fail with a validation error)
* per-version fields may not all be set to identical values (top-level field should be used instead)

v1 structure:
* per-version fields only

converting from v1beta1 -> v1:
* if per-version fields are set in v1beta1, use them for v1
* if top-level fields are set in v1beta1, copy them into each version in v1

converting from v1 -> v1beta1:
* if per-version fields are all identical, set the top-level field in v1beta1
* if per-version fields are not all identical, set the per-version fields in v1beta1

alternatives considered (and explanations why they were not chosen):
* top-level + per-version override
* auto-clearing per-version fields if top-level fields are set